tests/unit/sagemaker/image_uris/test_dlc_frameworks.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+from packaging.version import Version
+
+from sagemaker import image_uris
+
+ALTERNATE_DOMAINS = {
+    "cn-north-1": "amazonaws.com.cn",
+    "cn-northwest-1": "amazonaws.com.cn",
+    "us-iso-east-1": "c2s.ic.gov",
+}
+DOMAIN = "amazonaws.com"
+IMAGE_URI_FORMAT = "{}.dkr.ecr.{}.{}/{}:{}"
+INSTANCE_TYPES_AND_PROCESSORS = (("ml.c4.xlarge", "cpu"), ("ml.p2.xlarge", "gpu"))
+REGION = "us-west-2"
+
+DLC_ACCOUNT = "763104351884"
+DLC_ALTERNATE_REGION_ACCOUNTS = {
+    "ap-east-1": "871362719292",
+    "cn-north-1": "727897471807",
+    "cn-northwest-1": "727897471807",
+    "me-south-1": "217643126080",
+    "us-gov-west-1": "442386744353",
+    "us-iso-east-1": "886529160074",
+}
+SAGEMAKER_ACCOUNT = "520713654638"
+SAGEMAKER_ALTERNATE_REGION_ACCOUNTS = {
+    "ap-east-1": "057415533634",
+    "cn-north-1": "422961961927",
+    "cn-northwest-1": "423003514399",
+    "me-south-1": "724002660598",
+    "us-gov-west-1": "246785580436",
+    "us-iso-east-1": "744548109606",
+}
+
+
+def _expected_uri(repo, fw_version, account, py_version=None, processor="cpu", region=REGION):
+    domain = ALTERNATE_DOMAINS.get(region, DOMAIN)
+    tag = "{}-{}".format(fw_version, processor)
+    if py_version:
+        tag = "-".join((tag, py_version))
```

Comment:
I've seen this block of code multiple places. Refactor into function?

Reply:
I was trying to avoid calling the same code that's being tested, if that makes sense. I know it results in a lot of duplicated code in the tests, but otherwise I feel like the test wouldn't be doing its job 🤷‍♀️

Reply:
agreed